cyto_dl/nn/head/gan_head.py

```diff
@@ -80,13 +77,18 @@ def run_head(
                 )
             loss_D, loss_G = self._calculate_loss(y_hat, batch, discriminator)
 
-        y_hat_out, y_out = None, None
-        if save_image:
-            y_hat_out, y_out = self.save_image(y_hat, batch, stage)
-
         return {
             "loss_D": loss_D,
             "loss_G": loss_G,
-            "y_hat_out": y_hat_out,
-            "y_out": y_out,
+            "pred": self._postprocess(y_hat, img_type="prediction", n_postprocess=n_postprocess),
+            "target": self._postprocess(
+                batch[self.head_name], img_type="input", n_postprocess=n_postprocess
+            )
+            if stage != "predict"
+            else None,
+            "input": self._postprocess(
+                batch[self.x_key], img_type="input", n_postprocess=n_postprocess
```

Comment:
what is "img_type" here?

Reply:
in the config I broke down postprocesssing by whether it's applied to the input/target/prediction